techniques/ios/MASTG-TECH-0066.md

```diff
@@ -6,3 +6,109 @@ platform: ios
+
+```bash
+f~canEvaluatePolicy
+0x100008360 24 str.canEvaluatePolicy:error:
+```
+
+In the output, you can see a flag `str.canEvaluatePolicy:error:` representing the string `canEvaluatePolicy:error:` which is located at `0x100008360`. You can then use the [`axt` command](https://book.rada.re/analysis/code_analysis.html#analyze-xrefs) to find cross-references to this string using its address:
+
+```bash
+axt @ 0x100008360
+
+(nofunc) 0x100000350 [UNKNOWN] invalid
+sym.func.100004ea8 0x100004f10 [STRING] ldr x1, [x8, 0xa0]
+```
+
+The output shows the cross-references to the given address, indicating where the function is used in the binary in two different locations:
+
+- `0x100000350`: This indicates an invalid reference, possibly due to dynamic resolution.
+- `sym.func.100004ea8`: This indicates a valid reference to a function located at `0x100004ea8`, which can be further analyzed. The reference to the string will be at `0x100004f10`.
+
+### Identifying Strings
+
+**Listing Strings**:
+
+Use the [`iz` command](https://book.rada.re/tools/rabin2/strings.html) to list all strings in the binary. Combine it with [`~` filtering](https://book.rada.re/first_steps/syntax.html#filtering) to filter for specific strings. For example, to search for a string containing `somestring`:
+
+```bash
+iz~+somestring
+```
+
+Alternatively, you can use `izz`
+
+**Searching for Specific Strings**:
+
+Use the [`/` command](https://book.rada.re/search/basic_searches.html) to search for a string directly in the binary:
+
+```bash
+/ somestring
+```
+
+Use advanced search options with [`/`](https://book.rada.re/search/intro.html).
+
+### Analyzing Cross-References
+
+**Finding Cross-References**:
+
+Use the [`axt` command](https://book.rada.re/analysis/code_analysis.html#recursive-analysis) to find cross-references to a specific function. This helps understand how and where the function is invoked.
+
+```bash
+axt @ <function_address>
+```
+
+Note: The `@` symbol is used for [seeking to specific addresses](https://book.rada.re/commandline/seeking.html) in radare2.
+
+Example: In @MASTG-DEMO-0011, cross-references to `SecKeyCreateRandomKey` are analyzed to ensure secure key sizes.
+
+### Inspecting the Disassembly
+
+**Inspecting Function Calls**:
+
+Use the [`pdf` command](https://book.rada.re/commandline/print_modes.html) to decompile a function and inspect its implementation. This provides insights into how the function operates.
+
+```bash
+pdf @ <function_address>
+```
+
+The `@` symbol [seeks to the specified address](https://book.rada.re/commandline/seeking.html) before executing the command.
+
+**Inspecting Specific Instructions**:
+
+Use the [`pd` command](https://book.rada.re/commandline/print_modes.html) to disassemble a specific number of instructions at a given address. For example, to disassemble 5 instructions:
+
+```bash
+pd-- 5 @ 0x1000048c4
```

Comment:
the pd-- command shows 5 instructions before and 5 instructions after the current address. so maybe better to describe it this way.

Reply:
if you want to disassemble 5 instructions starting there, just use "pd 5 ..." without the --